ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/MacOS.yml

```diff
@@ -0,0 +1,563 @@
+  
+  # NEED TO FIGURE THIS OUT 
+
+  #- name: CPAN initialization
+  #  shell: | 
+  #    yes | cpan install
+
+  # NEED TO FIGURE THIS OUT 
+
+  - name: Install JSON
+    cpanm: 
+      executable: /usr/local/bin/cpan
+      name: JSON
+    tags:
+      - cpan
+      - json
+      - skip_ansible_lint
+    #become: yes
+  
+  - name: Install Text::CSV
+    cpanm: 
+      executable: /usr/local/bin/cpan
+      name: Text::CSV
+    tags:
+      - cpan
+      - text_csv
+      - skip_ansible_lint
+    #become: yes
+   
+  - name: Install XML::Parser
+    cpanm: 
+      executable: /usr/local/bin/cpan
+      name: XML::Parser
+    tags:
+      - cpan
+      - xml_parser
+      - skip_ansible_lint
+    #become: yes
+  
+  ###################################
+  # freemarker to Jenkins User home #
+  ###################################
+  - name: Check that the freemarker.jar exists
+    stat:
+      path: /Users/jenkins/freemarker.jar
+    register: freemarker
+    tags: freemarker
+  
+  # Originally downloaded from
+  # https://sourceforge.net/projects/freemarker/files/freemarker/2.3.8/freemarker-2.3.8.tar.gz
+  - name: Download and extract freemarker.jar
+    unarchive:
+      src: https://ci.adoptopenjdk.net/userContent/freemarker-2.3.8.tar.gz
+      dest: /tmp/
+      remote_src: yes
+      #mode: 0755
+    when: freemarker.stat.exists == False
+    become: yes
+    tags: freemarker
+  
+  - name: Move freemarker.jar to /home/{{ Jenkins_Username }} folder
+    command: mv /tmp/freemarker-2.3.8/lib/freemarker.jar /Users/jenkins
+    when: freemarker.stat.exists == False
+    become: yes
+    tags: freemarker
+  
+  #####################
+  # Bootjdk Directory #
+  #####################
+  
+  - name: Download for openjdk8
+    unarchive:
+      src: https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u192-b12/OpenJDK8U-jdk_x64_mac_hotspot_8u192b12.tar.gz
+      dest: /Users/jenkins
+      validate_certs: no
+      remote_src: yes
+    tags: jdk8
+  
+  - name: Create bootjdk directory
+    file: 
+      path: /Users/jenkins/bootjdks 
+      state: present
+  
+  - name: Move to proper directory
+    shell: |
+      mv /Users/jenkins/jdk8u192-b12/Contents/Home /Users/jenkins/bootjdks/adoptojdk-java-8
+    tags: jdk8
+  
+  # May need become: root for this one
+  - name: Cleanup jdk8 directories
+    shell: |
+      rm -rf /Users/jenkins/OpenJDK8U-jdk_x64_mac_hotspot_8u192b12.tar.gz jdk8u192-b12
+    become: yes
+    tags: jdk8
+  
+  ############################
+  # Additional Brew Packages #
+  ############################
+  
+  #- name: Install ant and ant-contrib
+  - homebrew: 
+      name: ant
+    #  - ant
+    #  - ant-contrib
+      state: present
+    tags: brew_packages
+  
+  ##############
+  # JDK11 Only #
+  ##############
+  - name: Download and extract JDK11 
+    unarchive:
+      src: https://github.com/AdoptOpenJDK/openjdk10-releases/releases/download/jdk-10.0.2%2B13/OpenJDK10_x64_Mac_jdk-10.0.2.13.tar.gz
```

Comment:
Try how the unix playbook role does this.
https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/8be6072a398346ab2d97a3dc352fe3c4ee44ce62/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk11/tasks/main.yml#L50